### PR TITLE
Managed instance data classification REST API - examples files name fixes

### PR DESCRIPTION
**Pull request by igsiroti**

### CODE REVIEW

**Comment by annatisch** [comment]
Thanks @igsiroti!
The changes look fine to me - however there are number of warnings in the breaking change CI that appear inconsistent with the diff in this PR.
Are you able to shed some light on these?

```
{
  "id": "1005",
  "code": "RemovedPath",
  "message": "The new version is missing a path that was found in the old version. Was path '/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/servers/{serverName}/databases/{databaseName}/sensitivityLabels' removed or restructured?",
  "jsonref": "#/paths/~1subscriptions~1{subscriptionId}~1resourceGroups~1{resourceGroupName}~1providers~1Microsoft.Sql~1servers~1{serverName}~1databases~1{databaseName}~1sensitivityLabels",
  "json-path": "#/paths/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/servers/{serverName}/databases/{databaseName}/sensitivityLabels",
  "type": "Warning"
}

{
  "id": "1005",
  "code": "RemovedPath",
  "message": "The new version is missing a path that was found in the old version. Was path '/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/servers/{serverName}/databases/{databaseName}/sensitivityLabels/{sensitivityLabelSource}' removed or restructured?",
  "jsonref": "#/paths/~1subscriptions~1{subscriptionId}~1resourceGroups~1{resourceGroupName}~1providers~1Microsoft.Sql~1servers~1{serverName}~1databases~1{databaseName}~1sensitivityLabels~1{sensitivityLabelSource}",
  "json-path": "#/paths/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/servers/{serverName}/databases/{databaseName}/sensitivityLabels/{sensitivityLabelSource}",
  "type": "Warning"
}

{
  "id": "1005",
  "code": "RemovedPath",
  "message": "The new version is missing a path that was found in the old version. Was path '/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/servers/{serverName}/databases/{databaseName}/schemas/{schemaName}/tables/{tableName}/columns/{columnName}/sensitivityLabels/{sensitivityLabelSource}' removed or restructured?",
  "jsonref": "#/paths/~1subscriptions~1{subscriptionId}~1resourceGroups~1{resourceGroupName}~1providers~1Microsoft.Sql~1servers~1{serverName}~1databases~1{databaseName}~1schemas~1{schemaName}~1tables~1{tableName}~1columns~1{columnName}~1sensitivityLabels~1{sensitivityLabelSource}",
  "json-path": "#/paths/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/servers/{serverName}/databases/{databaseName}/schemas/{schemaName}/tables/{tableName}/columns/{columnName}/sensitivityLabels/{sensitivityLabelSource}",
  "type": "Warning"
}

{
  "id": "1038",
  "code": "AddedPath",
  "message": "The new version is adding a path that was not found in the old version.",
  "jsonref": "#/paths/~1subscriptions~1{}~1resourceGroups~1{}~1providers~1Microsoft.Sql~1managedInstances~1{}~1databases~1{}~1schemas~1{}~1tables~1{}~1columns~1{}~1sensitivityLabels~1{}",
  "json-path": "#/paths/subscriptions/{}/resourceGroups/{}/providers/Microsoft.Sql/managedInstances/{}/databases/{}/schemas/{}/tables/{}/columns/{}/sensitivityLabels/{}",
  "type": "Info"
}

{
  "id": "1038",
  "code": "AddedPath",
  "message": "The new version is adding a path that was not found in the old version.",
  "jsonref": "#/paths/~1subscriptions~1{}~1resourceGroups~1{}~1providers~1Microsoft.Sql~1managedInstances~1{}~1databases~1{}~1sensitivityLabels",
  "json-path": "#/paths/subscriptions/{}/resourceGroups/{}/providers/Microsoft.Sql/managedInstances/{}/databases/{}/sensitivityLabels",
  "type": "Info"
}

{
  "id": "1038",
  "code": "AddedPath",
  "message": "The new version is adding a path that was not found in the old version.",
  "jsonref": "#/paths/~1subscriptions~1{}~1resourceGroups~1{}~1providers~1Microsoft.Sql~1managedInstances~1{}~1databases~1{}~1sensitivityLabels~1{}",
  "json-path": "#/paths/subscriptions/{}/resourceGroups/{}/providers/Microsoft.Sql/managedInstances/{}/databases/{}/sensitivityLabels/{}",
  "type": "Info"
}

{
  "id": "1007",
  "code": "RemovedClientParameter",
  "message": "The new version is missing a client parameter that was found in the old version. Was 'ServerNameParameter' removed or renamed?",
  "jsonref": "#/parameters",
  "json-path": "#/parameters",
  "type": "Warning"
}
```

**Comment by igsiroti** [comment]
Thanks @annatisch for the review! In the removed files there was a typo in their names '..Sensitivty...' instead of '..Sensitiv**i**ty...' so I renamed them (it shows as if removed and added) and updated the refs. Does it makes sense?

**Comment by annatisch** [comment]
Thanks @igsiroti - yes I understand the changes in the PR.
My question was more regarding the warnings I printed in my last post - it's showing path changes, for example:
`"#/paths/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/servers/{serverName}/databases/{databaseName}/sensitivityLabels"`
has been changed to:
`"#/paths/subscriptions/{}/resourceGroups/{}/providers/Microsoft.Sql/managedInstances/{}/databases/{}/sensitivityLabels"`

And the `serverName` client parameter used in the original URL is no longer present.

Were these path changes recently introduced to the spec? Just trying to figure out where these warnings are coming from :)

**Comment by igsiroti** [comment]
sorry my bad. adding it now

**Comment by annatisch** [comment]
Thanks @igsiroti - though I'm a bit confused - are these new changes to the API? Were they existing changes that were somehow reverted?

**Comment by annatisch** [comment]
Looking at the changes it appears that:
- Some new breaking APIs that were added to sensitivityLabels.json have been reverted which resolves the breaking changes.
- These new APIs have now been moved to managedInstanceSensitivityLabels.json and will coexist.

Does this sound accurate?
Can you confirm whether these new managedInstanceSensitivityLabels APIs have already received ARM signoff and if so could you please link to the PR?

Finally - are you aware that currently no SDKs are being generated for these new APIs? If you wish for them to be added to the SDK, please add the new spec to the package tag in the readme :)

Thanks!

**Comment by igsiroti** [comment]
- That is correct

- The APIs haven't received ARM signoff yet

- Yes I'm aware of that

Thanks

**Comment by annatisch** [comment]
Thanks @igsiroti!
Adding @ravbhatnagar for ARM review.

**Comment by annatisch** [comment]
Ping @ravbhatnagar

**Comment by annatisch** [comment]
Pinging @ravbhatnagar again - could you please take a look at the new [managed instance sensitivity label APIs](https://github.com/Azure/azure-rest-api-specs/pull/3673/files#diff-bbe31a36683b8da9c6a9763ee50b5e25)?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/managedInstanceSensitivityLabels.json
Comment (ravbhatnagar) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/managedInstanceSensitivityLabels.json]: schema for the default response much be the ARM error contract as described in RPC. Check Batch RP swagger for a sample.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/managedInstanceSensitivityLabels.json
Comment (ravbhatnagar) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/managedInstanceSensitivityLabels.json]: Should this be an enum with fixed set of values?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/managedInstanceSensitivityLabels.json
Comment (ravbhatnagar) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/managedInstanceSensitivityLabels.json]: schema, tables and columns are not modeled through the RP control plane. And you are exposing a nested resource type under data plane entities. This can potentially create a  gap in the experience. Can these three be modeled as properties you take as input on create instead of having them in the path? For GET, if user wants to get sensitivityLabel for a specific schema/table/column, they can use filter query param?
And then you can model sensitivityLabel directly under the database resource?

**Comment by ravbhatnagar** [comment]
@igsiroti FYI

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/managedInstanceSensitivityLabels.json
Comment (ravbhatnagar) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/managedInstanceSensitivityLabels.json]: checked with Igal. He will follow up with Jared since all SQL swaggers are generated by tool.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/managedInstanceSensitivityLabels.json
Comment (ravbhatnagar) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/managedInstanceSensitivityLabels.json]: Chatted with Igal and although it doesnt seem ideal but since other existing APIs are already using similar pattern, I will sign off on this. IN the meantime, Igal will discuss within his team if they would like to re-design these for a newer api-version, or GA version.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/managedInstanceSensitivityLabels.json
Comment (igsiroti) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/managedInstanceSensitivityLabels.json]: Information types may be managed by user. It's open set - not enum

**Comment by annatisch** [comment]
Thanks @ravbhatnagar 
@igsiroti - what is the status of this PR?
Are you waiting on input from a 3rd party regarding the ARM error responses in the spec?

**Comment by igsiroti** [comment]
Correct. I want to make sure we are OK with the default response @ravbhatnagar mentioned above. Waiting for Jared's response

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/managedInstanceSensitivityLabels.json
Comment (annatisch) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/managedInstanceSensitivityLabels.json]: The description says this is optional, but it's marked as required.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/managedInstanceSensitivityLabels.json
Comment (annatisch) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/managedInstanceSensitivityLabels.json]: Just to confirm the behavior:
For the GET and LIST operations, this can have a value of "current" or "recommended"
For the PUT and DELETE operations, this can *only* have a value of "current"

Is that correct?

It's worth noting that required enums of a single value (like "current") are treated as a constant in the generated SDKs and are therefore not exposed to the user and are simply added by default in the client.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/managedInstanceSensitivityLabels.json
Comment (igsiroti) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/managedInstanceSensitivityLabels.json]: Not sure how but it's old file. Fixing it

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/managedInstanceSensitivityLabels.json
Comment (igsiroti) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/managedInstanceSensitivityLabels.json]: All correct beside LIST. There the API is 
../databases/{databaseName}/sensitivityLabels
../databases/{databaseName}/currentSensitivityLabels
../databases/{databaseName}/recommendedSensitivityLabels

the sensitivityLabelSource parameter is removed there

**Comment by annatisch** [comment]
Thanks @igsiroti - it's looking good.
Last thing - You changed the names of the operations to "ManagedDatabaseSensitivityLabels" yes the spec itself is still called "ManagedInstanceSensitivityLabels.json" - could this be renamed?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/managedInstanceSensitivityLabels.json
Comment (annatisch) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/managedInstanceSensitivityLabels.json]: You're right - looks like I didn't have the latest changes :)

**Comment by igsiroti** [comment]
Thanks @annatisch. I renamed the spec

**Review by annatisch — APPROVED** [review]

**Pull request merged**
